public/video-ui/src/pages/Upload/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { createAsset } from '../../actions/VideoActions/createAsset';
+import { getVideo } from '../../actions/VideoActions/getVideo';
+import { revertAsset } from '../../actions/VideoActions/revertAsset';
+import { saveVideo } from '../../actions/VideoActions/saveVideo';
+import PlutoProjectLink from '../../components/Pluto/PlutoProjectLink';
+import { PlutoProjectPicker } from '../../components/Pluto/PlutoProjectPicker';
+import AddAssetFromURL from '../../components/VideoUpload/AddAssetFromURL';
+import AddSelfHostedAsset from '../../components/VideoUpload/AddSelfHostedAsset';
+import { VideoTrail } from '../../components/VideoUpload/VideoTrail';
+import YoutubeUpload from '../../components/VideoUpload/YoutubeUpload';
+import { startVideoUpload } from '../../slices/s3Upload';
+import { fetchCategories, fetchChannels } from '../../slices/youtube';
+import { AppDispatch, RootState } from '../../util/setupStore';
+import { getStore } from '../../util/storeAccessor';
+
+export const VideoUpload = (props: { params: { id: string } }) => {
+  const dispatch = useDispatch<AppDispatch>();
+
+  const store = useSelector(
+    ({
+      config,
+      error,
+      formFieldsWarning,
+      search,
+      s3Upload,
+      usage,
+      videoEditOpen,
+      youtube,
+      video,
+      uploads
+    }: RootState) => ({
+      config,
+      error,
+      formFieldsWarning,
+      search,
+      s3Upload,
+      usage,
+      videoEditOpen,
+      youtube,
+      video: video.video,
+      uploads,
+      activatingAssetNumber: video.activatingAssetNumber
+    })
+  );
+
+  useEffect(() => {
+    dispatch(getVideo(props.params.id));
+    if (store.youtube.categories.length === 0) {
+      dispatch(fetchCategories());
+    }
+    if (store.youtube.channels.length === 0) {
+      dispatch(fetchChannels());
+    }
+  }, [
+    dispatch,
+    props.params.id,
+    store.youtube.categories,
+    store.youtube.channels
+  ]);
+
+  const uploading = store.s3Upload.status === 'uploading';
```

Comment:
Small thing, but `isUploading` might be might the variable slightly clearer that it's a boolean

Reply:
good call 👍